docs/essentials/widget_language.md

```diff
@@ -334,28 +334,27 @@ finally {
 
 Let’s review this `finally` block in detail.
 
-`finally` is the keyword of the block, all statements wrapped by `{}` it as part of the block.
+`finally` is the keyword of the block, all statements wrapped by `{}` are part of the block.  
+In this block, there are two statements. Let's look deeper in them one by one.
 
-In this block, there are two statements. Let deep in one by one.
+The first statement `submit.tap_stream()`, returns the `tap` event as a ReactiveX `Observable` stream. Then we subscribe to it to update `greet`s text and then we reset `input`. The name of `unsubscribe_when_dropped` should be self-describing. It converts the handle to a variable that will result in auto-`unsubscribe` when it is dropped. So the whole statement subscribes to `submit`s`tap` and creates a variable to manage the subscriptions lifetime.
```

Comment:
It should be :


The next statement `move_to_widget!(guard);` is very simple. It transfers the ownership of  `guard` to `greet`, then the `guard` will live as long as `greet`.